### PR TITLE
docs: Better credits and contribution guide

### PR DESCRIPTION
**Pull request by eriksvedang**
Maintaining the contributors list seemed like a chore when Github already supports that feature in such a nice way.

More ideas for what should go into Contributing.md are most welcome!

### CODE REVIEW

File: docs/Contributing.md [hunk 1]
```diff
@@ -1,16 +1,15 @@
...
-Community
----------
+### Community
 The best place to start is to join the Carp Gitter channel over at
 [https://gitter.im/carp-lang/Carp](https://gitter.im/carp-lang/Carp).
 
+### Understanding the Compiler
+See [Hacking.md](hacking.md) for a gentle introduction to the Carp compiler internals.
 
 License
```
Comment (hellerve) [inline docs/Contributing.md:14]: Should we also change this heading, then?

**Review by hellerve — APPROVED** [review]

File: docs/Contributing.md (hunk 1 above)
Comment (eriksvedang) [reply docs/Contributing.md:14]: You mean Hacking.md becomes UnderstandingTheCompiler.md ?

Comment (hellerve) [reply docs/Contributing.md:14]: No, this was on the `License` heading which is still the old MD style.

Comment (eriksvedang) [reply docs/Contributing.md:14]: Now they all use two '#' (if that's what you meant)

**Pull request merged**